sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -1493,6 +1493,8 @@ version
 operatorPipeRightSide
     : selectClause
     | whereClause
+    | pivotClause? unpivotClause?
```

Comment:
QQ: why can't be simply
```
| pivotClause
| unpivotClause
```
here?

Reply:
Good question. I originally did that, but we wanted to return descriptive error messages if the query erroneously includes both the PIVOT and UNPIVOT clauses consecutively (this is not allowed).

I updated this to

```
    | pivotClause unpivotClause?
    | unpivotClause pivotClause?
```

and added a comment

```
// The following two cases match the PIVOT or UNPIVOT clause, respectively.
// For each one, we add the other clause as an option in order to return high-quality
// error messages in the event that both are present (this is not allowed).
```

Hopefully this helps 👍